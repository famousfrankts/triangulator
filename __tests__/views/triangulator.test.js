import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Triangulator from '../../src/views/triangulator';

const mockStore = configureStore();

describe('<Triangulator />', () => {
  describe('Rendering', () => {
    test ('it renders the component', () => {
      const store = mockStore({
        triangle: { a: 10, b: 10, c: 10 },
      });
      const component = mount(<Triangulator store={store} />);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
