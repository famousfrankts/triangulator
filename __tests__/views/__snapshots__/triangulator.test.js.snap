// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Triangulator /> Rendering it renders the component 1`] = `
<Connect(Triangulator)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Triangulator
    setA={[Function]}
    setB={[Function]}
    setC={[Function]}
    setTriangle={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    triangle={
      Object {
        "a": 10,
        "b": 10,
        "c": 10,
      }
    }
  >
    <div
      className="triangulator"
    >
      <section
        className="description"
      >
        <h1>
          Triangulator
        </h1>
        <p>
          Welcome to the triangulator! The Triangulator is capable of many things. Predicting the future. Changing the spin of the earth. Calculating the meaning of life. But most importantly, figuring out what type of triangle you get by declaring its 3 sides lengths:
        </p>
      </section>
      <form
        data-ts="Form"
      >
        <fieldset>
          <label
            className=""
          >
            <span>
              Side A
            </span>
            <input
              min="1"
              onChange={[Function]}
              type="number"
              value={10}
            />
          </label>
        </fieldset>
        <fieldset>
          <label
            className=""
          >
            <span>
              Side B
            </span>
            <input
              min="1"
              onChange={[Function]}
              type="number"
              value={10}
            />
          </label>
        </fieldset>
        <fieldset>
          <label
            className=""
          >
            <span>
              Side C
            </span>
            <input
              min="1"
              onChange={[Function]}
              type="number"
              value={10}
            />
          </label>
        </fieldset>
      </form>
      <div
        data-ts="Note"
      >
        <i
          className="ts-icon-info"
        />
        <p>
          After careful consideration, we have determined that the triangle is 
          !
        </p>
      </div>
    </div>
  </Triangulator>
</Connect(Triangulator)>
`;
